src/java/org/apache/cassandra/io/util/CompressedChunkReader.java

```diff
@@ -83,20 +89,38 @@ public BufferType preferredBufferType()
     }
 
     @Override
-    public Rebufferer instantiateRebufferer()
+    public Rebufferer instantiateRebufferer(boolean isScan)
     {
-        return new BufferManagingRebufferer.Aligned(this);
+        return new BufferManagingRebufferer.Aligned(this.setMode(isScan));
     }
 
     public static class Standard extends CompressedChunkReader
     {
         // we read the raw compressed bytes into this buffer, then uncompressed them into the provided one.
         private final ThreadLocalByteBufferHolder bufferHolder;
+        private final ThreadLocalReadAheadBuffer readAheadBuffer;
+
 
         public Standard(ChannelProxy channel, CompressionMetadata metadata, Supplier<Double> crcCheckChanceSupplier)
         {
             super(channel, metadata, crcCheckChanceSupplier);
             bufferHolder = new ThreadLocalByteBufferHolder(metadata.compressor().preferredBufferType());
+
+            int readAheadBufferSize = DatabaseDescriptor.getCompressedReadAheadBufferSize();
+            readAheadBuffer = readAheadBufferSize > 0 ? new ThreadLocalReadAheadBuffer(channel, readAheadBufferSize, metadata.compressor().preferredBufferType()) : null;
+        }
+
+        @Override
+        public CompressedChunkReader setMode(boolean isScan)
```

Comment:
nit: style guide recommends to avoid `boolean` in favor of enums to help make the code more expressive

Reply:
Will create a ReadMode enum in `CompressedChunkReader`

Reply:
How strongly do you feel about this? I would have to plumb it through several layers (up to `RebuffererFactory`)